common/flagdinjector/flagdinjector.go

```diff
@@ -195,40 +193,49 @@ func (fi *FlagdContainerInjector) buildSources(ctx context.Context, objectMeta *
 			source.Provider = flagSourceConfig.DefaultSyncProvider
 		}
 
-		var sourceCfg types.SourceConfig
-		var err error
-
-		switch {
-		case source.Provider.IsKubernetes():
-			sourceCfg, err = fi.toKubernetesProviderConfig(ctx, objectMeta, podSpec, source)
-			if err != nil {
-				return []types.SourceConfig{}, err
-			}
-		case source.Provider.IsFilepath():
-			sourceCfg, err = fi.toFilepathProviderConfig(ctx, objectMeta, podSpec, sidecar, source)
-			if err != nil {
-				return []types.SourceConfig{}, err
-			}
-		case source.Provider.IsHttp():
-			sourceCfg = fi.toHttpProviderConfig(source)
-		case source.Provider.IsGrpc():
-			sourceCfg = fi.toGrpcProviderConfig(source)
-		case source.Provider.IsFlagdProxy():
-			sourceCfg, err = fi.toFlagdProxyConfig(ctx, objectMeta, source)
-			if err != nil {
-				return []types.SourceConfig{}, err
-			}
-		default:
-			return []types.SourceConfig{}, fmt.Errorf("could not add provider %s: %w", source.Provider, constant.ErrUnrecognizedSyncProvider)
+		sourceCfg, err := fi.createSourceConfig(ctx, source, objectMeta, podSpec, sidecar)
+		if err != nil {
+			return []types.SourceConfig{}, err
 		}
 
-		sourceCfgCollection = append(sourceCfgCollection, sourceCfg)
+		sourceCfgCollection = append(sourceCfgCollection, *sourceCfg)
 
 	}
 
 	return sourceCfgCollection, nil
 }
 
+func (fi *FlagdContainerInjector) createSourceConfig(ctx context.Context, source api.Source, objectMeta *metav1.ObjectMeta, podSpec *corev1.PodSpec, sidecar *corev1.Container) (*types.SourceConfig, error) {
+	var sourceCfg types.SourceConfig
+	var err error
+
+	switch {
+	case source.Provider.IsKubernetes():
+		sourceCfg, err = fi.toKubernetesProviderConfig(ctx, objectMeta, podSpec, source)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
we can remove these returns of the errors and change the last one to be `return &sourceCfg, err`